File: 2022/02/day02.py
#!/usr/bin/python3
import sys
from typing import List


def part_one(lines: List[str]):
    print('Part One')

    result = 0
    for line in lines:
        plays = line.split(' ')

        if plays[1] == 'X': # Rock
            result += 1
            if plays[0] == 'A': # Rock
                result += 3
            elif plays[0] == 'B': # Paper
                result += 0
            else: # Scissors
                result += 6
        elif plays[1] == 'Y': # Paper
            result += 2
            if plays[0] == 'A': # Rock
                result += 6
            elif plays[0] == 'B': # Paper
                result += 3
            else: # Scissors
                result += 0
        else: # Scissors
            result += 3
            if plays[0] == 'A': # Rock
                result += 0
            elif plays[0] == 'B': # Paper
                result += 6
            else: # Scissors
                result += 3

    print(f'Result: {result}')


def part_two(lines: List[str]):
    print('Part Two')

    result = 0
    for line in lines:
        plays = line.split(' ')
        if plays[1] == 'X': # Lose
            result += 0
            if plays[0] == 'A': # Rock / Scissors
                result += 3
            elif plays[0] == 'B': # Paper / Rock
                result += 1
            else: # Scissors / Paper
                result += 2
        elif plays[1] == 'Y': # Draw
            result += 3
            if plays[0] == 'A': # Rock / Rock
                result += 1
            elif plays[0] == 'B': # Paper / Paper
                result += 2
            else: # Scissors/ Scissors
                result += 3
        else: # Win
            result += 6
            if plays[0] == 'A': # Rock / Paper
                result += 2
            elif plays[0] == 'B': # Paper / Scissors
                result += 3
            else: # Scissors / Rock
                result += 1

    print(f'Result: {result}')


def main(args):
    if len(args) < 2:
        print('[ERROR] Enter a file name!')
        print(f'\tUsage: python {args[0]} <filename>')
        sys.exit(1)

    filename = args[1]
    with open(filename, 'r') as f:
        lines = f.readlines()
        lines = [l.strip() for l in lines]

        part_one(lines)
        part_two(lines)


if __name__ == '__main__':
    main(sys.argv)

File: 2022/02/day02.go
package main

import (
    "bufio"
    "fmt"
    "log"
    "os"
)


func partOne(filename string) {
    fmt.Println("Part One ####################")

    // Read file
    file, err := os.Open(filename)
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    // Init result
    var result int = 0

    game_result := map[string]int{
        "A X": 1 + 3,
        "A Y": 2 + 6,
        "A Z": 3 + 0,
        "B X": 1 + 0,
        "B Y": 2 + 3,
        "B Z": 3 + 6,
        "C X": 1 + 6,
        "C Y": 2 + 0,
        "C Z": 3 + 3,
    }

    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        line := scanner.Text()
        // Code goes here

        result += game_result[line]
    }

    fmt.Printf("Result: %d\n", result)
}

func partTwo(filename string) {
    fmt.Println("Part Two ####################")

    // Read file
    file, err := os.Open(filename)
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    // Init result
    var result int = 0

    game_result := map[string]int{
        "A X": 0 + 3,
        "A Y": 3 + 1,
        "A Z": 6 + 2,
        "B X": 0 + 1,
        "B Y": 3 + 2,
        "B Z": 6 + 3,
        "C X": 0 + 2,
        "C Y": 3 + 3,
        "C Z": 6 + 1,
    }

    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        line := scanner.Text()
        // Code goes here

        result += game_result[line]
    }

    fmt.Printf("Result: %d\n", result)
}


func main() {
    if len(os.Args[1:]) < 1 {
        log.Fatal("Not enough arguments given!")
    }

    filename := os.Args[1]
    partOne(filename)
    partTwo(filename)
}
